### PR TITLE
Validate Root Container and add. args

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 1]
```diff
@@ -1314,3 +1314,14 @@ function Add-AzureVMCustomScriptExtension
 
     Write-Verbose "Successfully added the custom script extension '$extensionName' for virtual machine '$vmName'"
 }
+
+function Check-ContainerNameAndArgs
+{
+    param([string]$containerName,
+          [string]$additionalArguments)
+    
+    if($containerName -eq '$root' -and ($additionalArguments -eq "/S" -or $additionalArguments -like "/S *" -or $additionalArguments -like "* /S *" -or $additionalArguments -like "* /S"))
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/Utility.ps1]: Would it be better to use regular expression to match?

File: Tasks/AzureFileCopy/task.json [hunk 2]
```diff
@@ -323,6 +323,7 @@
         "AFC_AddNetworkSecurityRuleFailed": "Failed to add the network security rule: {0}",
         "AFC_UnableToSetCustomScriptExtension": "Unable to set the custom script extension '{0}' for virtual machine '{1}': {2}",
         "AFC_CopyPrereqsFailed": "Failed to enable copy prerequisites. {0}",
-        "AFC_BlobStorageNotFound": "Storage account: {0} not found. Please specify existing storage account"
+        "AFC_BlobStorageNotFound": "Storage account: {0} not found. Please specify existing storage account",
+        "AFC_RootContainerAndDirectory": "Root Container ($root) has no support for virtual directories. Remove '/S' from additional arguments."
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/task.json]: Review this message with @RoopeshNair

File: Tasks/AzureFileCopy/Utility.ps1 (hunk 1 above)
Comment (KrishnaAdityaB) [inline Tasks/AzureFileCopy/Utility.ps1]: +1 for regex.
Won't */S* do?
and is there any difference between /s and /S?

Comment (vincentdass) [inline Tasks/AzureFileCopy/Utility.ps1]: thanks @KrishnaAdityaB , The /S is case-insensitive. Will add a regex which checks for both the cases

Comment (vincentdass) [inline Tasks/AzureFileCopy/Utility.ps1]: done

File: Tasks/AzureFileCopy/task.json (hunk 2 above)
Comment (vincentdass) [inline Tasks/AzureFileCopy/task.json]: done

File: Tasks/AzureFileCopy/AzureFileCopy.ps1 [hunk 3]
```diff
@@ -44,6 +44,7 @@ $isPremiumStorage = $false
 
 $sourcePath = $sourcePath.Trim('"')
 $storageAccount = $storageAccount.Trim()
+$containerName = $containerName.Trim().ToLower()
```
Comment (chshrikh) [inline Tasks/AzureFileCopy/AzureFileCopy.ps1]: $containerName is non mandatory parameter so cant assume it is not null. Also we have already a code at line no 105 to pre-process this object, better to keep this at one place

Comment (vincentdass) [inline Tasks/AzureFileCopy/AzureFileCopy.ps1]: We process the input in the first place and then apply logic to them.
In the first part, we process the input and in line 105, we apply logic to them.
We can have them seperately.

**Pull request merged**